components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -98,6 +99,13 @@ private void initializeDataPublisher() {
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
 
+        if (APIUtil.isAnalyticsEnabled()) {
+            WebSocketUtils.setApiPropertyToChannel(ctx, Constants.REQUEST_START_TIME_PROPERTY,
+                    System.currentTimeMillis());
+            // resets the property since context is shared
```

Comment:
```suggestion
            // Resets the property since context is shared. 
            // If this property is non-zero, it means that the frame is coming from backend, to client.
            // If not, it means that the frame is coming from client, to backend.
```